stdlib/public/Concurrency/Task.swift

```diff
@@ -472,6 +472,38 @@ public func _runAsyncHandler(operation: @escaping () async -> ()) {
   )
 }
 
+// ==== Voluntary Suspension -----------------------------------------------------
+extension Task {
+
+  /// Explicitly suspend the current task, potentially giving up execution actor
+  /// of current actor/task, allowing other tasks to execute.
+  ///
+  /// This is not a perfect cure for starvation;
+  /// if the task is the highest-priority task in the system, it might go
+  /// immediately back to executing.
+  public static func yield() async {
+    // Prepare the job flags
+    var flags = JobFlags()
+    flags.kind = .task
+    flags.priority = .default
+    flags.isFuture = true
+
+    // Create the asynchronous task future, it will do nothing, but simply serves
+    // as a way for us to yield our execution until the executor gets to it and
+    // resumes us.
+    // FIXME: This should be an empty closure instead. Returning `0` here is
+    //        a workaround for rdar://74957357
+    // TODO: consider if it would be useful for this task to be a child task
+    let (task, _) = Builtin.createAsyncTaskFuture(flags.bits, nil, { return 0 })
+
+    // Enqueue the resulting job.
+    _enqueueJobGlobal(Builtin.convertTaskToJob(task))
+
+    let _ = await Handle<Int, Never>(task).get()
+  }
```

Comment:
So this is a pretty naive implementation;

As with sleep() I am wondering if we really need to create a new task to do this?

See also: https://github.com/apple/swift/pull/35983#discussion_r576530459

Reply:
(I'm looking into implementing this without a new Task now)